azure/services/roleassignments/client.go

```diff
@@ -28,7 +32,12 @@ import (
 
 // client wraps go-sdk.
 type client interface {
-	Create(context.Context, string, string, authorization.RoleAssignmentCreateParameters) (authorization.RoleAssignment, error)
+	//Create(context.Context, string, string, authorization.RoleAssignmentCreateParameters) (authorization.RoleAssignment, error)
```

Comment:
Do we need this client interface? In the other async PRs, we've been able to remove them along with the client mocks if they're not directly called by a service.

Reply:
Will check on this part.

Reply:
I think you should be able to since it's not imported anywhere and it's not needed in your refactor of Reconcile() and Delete(). You could try removing it to verify that it doesn't break anything ,and make sure to remove client_mock.go and the references to it in doc.go.